### PR TITLE
move incl/phi/prior_std calculation in preprocess

### PR DESCRIPTION
**Pull request by mkanai**
I moved incl/phi/prior_std calculation to the preprocess task so that they won't be computed for every region.

@Fedja re: `is_binary` flag that we discussed, I don't know it's necessary because we now hard-coded 0/1 in L71 anyways. When binary phenotypes are coded other than 0/1/NA, is there any simple way to infer binary codings?

### CODE REVIEW

**Comment by Fedja** [comment]
> I moved incl/phi/prior_std calculation to the preprocess task so that they won't be computed for every region.
> 
> @Fedja re: `is_binary` flag that we discussed, I don't know it's necessary because we now hard-coded 0/1 in L71 anyways. When binary phenotypes are coded other than 0/1/NA, is there any simple way to infer binary codings?

it would be to protect the users from themselves but not necessary off course. is_binary would check that the values are 0,1,NA and nag if not.
say user accidentally codes cases/controls with 1/2 or happens to have one erroneus value there which triggers quantitative. At least could add bolded notification in readme that if there are values other than 1,0,NA quant pheno is assumed.

File: wdl/finemap.wdl [hunk 1]
```diff
@@ -30,7 +31,58 @@ task preprocess {
...
+                err = 1
+                exit 0
+            }
+        }
+        END {
+            if (!err) {
+                phi = vals["1"] / (vals["1"]+vals["0"])
+                var_y = phi * (1-phi)
+                std = 0.05 * sqrt(phi*(1-phi))
+                printf var_y > "var_y.txt"
```
Comment (Fedja) [inline wdl/finemap.wdl:74]: printf here and not others does not affect anything except missing newline but looks suspicious to autist like me :)

Comment (mkanai) [reply wdl/finemap.wdl]: great catch! addressed

**Comment by mkanai** [comment]
> > I moved incl/phi/prior_std calculation to the preprocess task so that they won't be computed for every region.
> > @Fedja re: `is_binary` flag that we discussed, I don't know it's necessary because we now hard-coded 0/1 in L71 anyways. When binary phenotypes are coded other than 0/1/NA, is there any simple way to infer binary codings?
> 
> it would be to protect the users from themselves but not necessary off course. is_binary would check that the values are 0,1,NA and nag if not. say user accidentally codes cases/controls with 1/2 or happens to have one erroneus value there which triggers quantitative. At least could add bolded notification in readme that if there are values other than 1,0,NA quant pheno is assumed.

I agree the flag would prevent mistakes but it also prevent fine-mapping binary and quantitative traits altogether. I added a bold note in README for now.

File: README.md [hunk 2]
```diff
@@ -19,7 +19,8 @@ Please run [`finemap.wdl`](wdl/finemap.wdl) on a cromwell server with an appropr
 
 Configurable options include:
 - `finemap.sumstats_pattern`: a path to GWAS summary statistics where `{PHENO}` is a magic keyword to be replaced by an actual phenotype name in `finemap.phenolistfile`
-- `finemap.phenotypes`: a list of phenotypes to fine-map
+- `finemap.phenolistfile`: a list of phenotypes to fine-map
+- `finemap.phenotypes`: a path to phenotype file. **[IMPORTANT]** The first column should contains sample IDs that match to bgen sample IDs. If a phenotype is binary, we expect it's coded as **0/1/NA**.
```
Comment (Fedja) [inline README.md:23]: continue like: ....coded 0/1/NA, otherwise quantitative phenotype is automatically assumed

**Review by Fedja — APPROVED** [review]

**Review by Lipastomies — APPROVED** [review]
Looking good! I agree with the assumption of a certain data form for binary data, as long as it's documented clearly it should not be a problem.

File: README.md (hunk 2 above)
Comment (mkanai) [reply README.md]: addressed!

**Pull request merged**